components/lib/dialog/Dialog.js

```diff
@@ -449,7 +449,8 @@ export const Dialog = React.forwardRef((inProps, ref) => {
                     type: 'button',
                     className: cx('closeButton'),
                     'aria-label': ariaLabel,
-                    onClick: onClose
+                    onClick: onClose,
+                    onKeyDown: (ev) => props.stopPropagationOnKeyDown && ev.stopPropagation()
```

Comment:
this doesn't need to be a property right? Shouldn't stopPropagation just be the default?

Reply:
Yeah it could be the default. I just added a property because I thought that maybe someone needs that the event propagates for some reason, this way the code adapts for different needs but maybe we can try to stop the propagation by default.